databox/api/tests/Api/StoryTest.php

```diff
@@ -0,0 +1,193 @@
+            'headers' => [
+                'Authorization' => $adminAuthorization,
+            ],
+        ]);
+        $this->assertResponseStatusCodeSame(404);
+    }
+
+    public function testRemoveStoryByAsset(): void
+    {
+        self::enableFixtures();
+
+        $adminAuthorization = 'Bearer '.KeycloakClientTestMock::getJwtFor(KeycloakClientTestMock::ADMIN_UID);
+
+        $client = static::createClient();
+        $collectionId = "";
+        $assetId = "";
+        $this->createStory($client, $assetId,$collectionId);
+
+        $client->request('DELETE', '/assets/' . urlencode($assetId), [
+            'headers' => [
+                'Authorization' => $adminAuthorization,
+            ],
+        ]);
+        $this->assertResponseStatusCodeSame(204);
+
+        $client->request('GET', '/assets/' . urlencode($assetId), [
+            'headers' => [
+                'Authorization' => $adminAuthorization,
+            ],
+        ]);
+        $this->assertResponseStatusCodeSame(404);
+
+        $client->request('GET', '/collections/' . urlencode($collectionId), [
+            'headers' => [
+                'Authorization' => $adminAuthorization,
+            ],
+        ]);
+        $this->assertResponseStatusCodeSame(404);
+    }
+
+    public function testCreateStoryManualLink(): void
+    {
+        $client = static::createClient();
+
+        $response = $client->request('POST', '/collections', [
+            'headers' => [
+                'Authorization' => 'Bearer '.KeycloakClientTestMock::getJwtFor(KeycloakClientTestMock::ADMIN_UID),
+            ],
+            'json' => [
+                // this title should become null when the collection becomes a storyCollection
+                'title' => 'Dummy story-collection',
+                'workspace' => $this->findIriBy(Workspace::class, [
+                    'slug' => 'test-workspace',
+                ])
+            ],
+        ]);
+        $this->assertResponseStatusCodeSame(201);
+        $collectionId = $response->toArray()['id'];
+        $collectionIri = $response->toArray()['@id'];
+
+        $response = $client->request('POST', '/assets', [
+            'headers' => [
+                'Authorization' => 'Bearer '.KeycloakClientTestMock::getJwtFor(KeycloakClientTestMock::ADMIN_UID),
+            ],
+            'json' => [
+                'title' => 'Dummy story-asset',
+                'workspace' => $this->findIriBy(Workspace::class, [
+                    'slug' => 'test-workspace',
+                ]),
+                'storyCollection' => $collectionIri,
+            ],
+        ]);
+        $this->assertResponseStatusCodeSame(201);
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $assetId = $response->toArray()['id'];
+
+        $this->checkRelation($client, $assetId, $collectionId);
+    }
+
+    private function createStory(Client $client, string &$assetId, string &$collectionId): void
```

Comment:
Make references optional:

```suggestion
    private function createStory(Client $client, ?string &$assetId = null, ?string &$collectionId = null): void
```